server/team_matchmaking_service.py

```diff
@@ -0,0 +1,117 @@
+from typing import NamedTuple
+
+from server import GameService
+from server.decorators import with_logger
+from server.lobbyconnection import ClientError
+from server.players import Player
+from server.team_matchmaker.player_party import PlayerParty
+
+MapDescription = NamedTuple('Map', [("id", int), ("name", str), ("path", str)])
+GroupInvite = NamedTuple('GroupInvite', [("inviting", Player), ("invited", Player), ("party", PlayerParty)])
+
+
+@with_logger
+class TeamMatchmakingService:
+    """
+    Service responsible for managing the global team matchmaking. Does grouping, matchmaking, updates statistics, and
+    launches the games.
+    """
+
+    def __init__(self, games_service: GameService):
+        self.game_service = games_service
+        self.player_parties = dict() # player -> current party
+        self._pending_invites = dict() # invited player -> current pending invite to this player
+
+    def invite_player_to_group(self, inviting_player: Player, invited_player: Player):
+        if inviting_player not in self.player_parties:
+            self.player_parties[inviting_player] = PlayerParty(self, inviting_player)
+
+        inviting_party = self.player_parties.get(inviting_player)
+
+        if not inviting_party.owner == inviting_player:
```

Comment:
I think `party` as a name suffices, given how the "inviting" bit is already inferred by there being an `inviting_player` above.

Reply:
It could also be the party the invited person is already in :)